core/Piranha/Web/ArchiveRouter.cs

```diff
@@ -24,163 +24,154 @@ public class ArchiveRouter
-                                    LastModified = blog.LastModified
-                                }
-                            };
+                            foundCategory = false;
+                        }
+                    }
+
+                    if (foundTag)
+                    {
+                        try
+                        {
+                            tagId = (await api.Posts.GetTagBySlugAsync(blog.Id, segments[n]).ConfigureAwait(false))?.Id ??
+                                    Guid.Empty;
+                        }
+                        finally
+                        {
+                            foundTag = false;
+                        }
+                    }
+
+                    if (foundPage)
+                    {
+                        try
+                        {
+                            page = Convert.ToInt32(segments[n]);
+                        }
+                        catch
+                        {
+                            // We don't care about the exception, we just
+                            // discard malformed input
+                        }
+                        break;
+                    }
+
+                    if (!year.HasValue)
+                    {
+                        try
+                        {
+                            year = Convert.ToInt32(segments[n]);
+
+                            if (year.Value > DateTime.Now.Year)
+                                year = DateTime.Now.Year;
+                        }
+                        catch
+                        {
+                            // We don't care about the exception, we just
+                            // discard malformed input
+                        }
+                    }
+                    else
+                    {
+                        try
+                        {
+                            month = Math.Max(Math.Min(Convert.ToInt32(segments[n]), 12), 1);
+                        }
+                        catch
+                        {
+                            // We don't care about the exception, we just
+                            // discard malformed input
                         }
                     }
                 }
+
+                return new RouteResponse
+                {
+                    PageId = blog.Id,
+                    Route = route,
+                    QueryString = $"id={blog.Id}&year={year}&month={month}&page={page}&pagenum={page}&category={categoryId}&tag={tagId}&piranha_handled=true",
+                    IsPublished = blog.Published.HasValue && blog.Published.Value <= DateTime.Now,
+                    CacheInfo = new HttpCacheInfo
+                    {
+                        EntityTag = Utils.GenerateETag(blog.Id.ToString(), blog.LastModified),
+                        LastModified = blog.LastModified
+                    }
+                };

```

Comment:
I'm pretty sure that you've only refactored the file to use initial return statements instead of nested if statements, but it is almost impossible to tell from the diff tool. If this is the only change please revert it as we try to stay away from changes that only is code refactoring as they are extremely hard to follow and adds an uncertainty in the commit history. If you have in fact introduced some other change, please specify!

Reply:
Half and half, they are just refactoring but those refactoring are there to fix codacy errors.
Just in case, i will revert the file, we are always in time to take it back :)